[task] Task server tests are broken

<!-- Please provide a detailed description of the bug. -->
### Bug Description:

<!-- Please provide clear steps to reproduce the bug. -->
### Steps to Reproduce:

1. Clone the repo
2. `yarn`
3. `yarn test`

```
  7 failing

  1) Task server / back-end
       task running in terminal - expected data is received from the terminal ws server:
     Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/akos.kitta/Desktop/theia/packages/task/lib/node/task-server.slow-spec.js)
  

  2) Task server / back-end
       task can successfully execute command found in system path using a terminal process:
     Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/akos.kitta/Desktop/theia/packages/task/lib/node/task-server.slow-spec.js)
  

  3) Task server / back-end
       shell task should execute the command as a whole if not arguments are specified:

      AssertionError: expected 127 to equal 0
      + expected - actual

      -127
      +0
      
      at Context.<anonymous> (lib/node/task-server.slow-spec.js:557:51)
      at step (lib/node/task-server.slow-spec.js:48:23)
      at Object.next (lib/node/task-server.slow-spec.js:29:53)
      at fulfilled (lib/node/task-server.slow-spec.js:20:58)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  4) Task server / back-end
       shell task should be able to exec using simple arguments:

      AssertionError: expected 127 to equal 0
      + expected - actual

      -127
      +0
      
      at Context.<anonymous> (lib/node/task-server.slow-spec.js:591:51)
      at step (lib/node/task-server.slow-spec.js:48:23)
      at Object.next (lib/node/task-server.slow-spec.js:29:53)
      at fulfilled (lib/node/task-server.slow-spec.js:20:58)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  5) Task server / back-end
       shell task should be able to run using arguments containing whitespace:

      AssertionError: expected 127 to equal 0
      + expected - actual

      -127
      +0
      
      at Context.<anonymous> (lib/node/task-server.slow-spec.js:608:51)
      at step (lib/node/task-server.slow-spec.js:48:23)
      at Object.next (lib/node/task-server.slow-spec.js:29:53)
      at fulfilled (lib/node/task-server.slow-spec.js:20:58)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  6) Task server / back-end
       shell task should be able to run using arguments specifying which quoting method to use:

      AssertionError: expected 127 to equal 0
      + expected - actual

      -127
      +0
      
      at Context.<anonymous> (lib/node/task-server.slow-spec.js:642:51)
      at step (lib/node/task-server.slow-spec.js:48:23)
      at Object.next (lib/node/task-server.slow-spec.js:29:53)
      at fulfilled (lib/node/task-server.slow-spec.js:20:58)
      at process._tickCallback (internal/process/next_tick.js:68:7)

  7) Task server / back-end
       shell task should be able to run using arguments with forbidden characters but no whitespace:

      AssertionError: expected 127 to equal 0
      + expected - actual

      -127
      +0
      
      at Context.<anonymous> (lib/node/task-server.slow-spec.js:659:51)
      at step (lib/node/task-server.slow-spec.js:48:23)
      at Object.next (lib/node/task-server.slow-spec.js:29:53)
      at fulfilled (lib/node/task-server.slow-spec.js:20:58)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

<!-- Please provide any additional information available. -->
<!-- Additional information can be in the form of logs, screenshots, screencasts. -->

### Additional Information

- Operating System:
- Theia Version: bb43e9e

--- a/.travis.yml
+++ b/.travis.yml
@@ -115,6 +115,7 @@ jobs:
     - stage: test
       os: linux
     - os: osx
+      osx_image: xcode11.4
       env: CXX=c++
       before_script: skip
       script:

--- a/packages/process/src/common/shell-command-builder.ts
+++ b/packages/process/src/common/shell-command-builder.ts
@@ -77,8 +77,6 @@ export class ShellCommandBuilder {
                 return this.buildForCmd(args, cwd, env);
             }
         }
-        // If we cannot detect which shell is being used, don't escape.
-        console.warn(`Unknown shell, could not escape arguments: ${host || 'undefined'}`);
         return this.buildForDefault(args, cwd, env);
     }
 
@@ -151,7 +149,7 @@ export class ShellCommandBuilder {
     }
 
     protected buildForDefault(args: Array<string | ShellQuotedString>, cwd?: string, env?: Array<[string, string | null]>): string {
-        return args.join(' ');
+        return this.buildForBash(args, cwd, env);
     }
 
 }

--- a/packages/task/src/node/process/process-task-runner.ts
+++ b/packages/task/src/node/process/process-task-runner.ts
@@ -190,6 +190,10 @@ export class ProcessTaskRunner implements TaskRunner {
             } else if (/(ps|pwsh|powershell)(.exe)?/.test(command)) {
                 quotingFunctions = PowershellQuotingFunctions;
                 execArgs = ['-c'];
+            } else {
+                quotingFunctions = BashQuotingFunctions;
+                execArgs = ['-l', '-c'];
+
             }
             // Allow overriding shell options from task configuration.
             if (shell && shell.args) {

--- a/packages/task/src/node/task-server.slow-spec.ts
+++ b/packages/task/src/node/task-server.slow-spec.ts
@@ -101,6 +101,9 @@ describe('Task server / back-end', function (): void {
         const taskInfo: TaskInfo = await taskServer.run(createProcessTaskConfig('shell', `${command} ${someString}`), wsRoot);
         const terminalId = taskInfo.terminalId;
 
+        const messagesToWaitFor = 10;
+        const messages: string[] = [];
+
         // hook-up to terminal's ws and confirm that it outputs expected tasks' output
         await new Promise((resolve, reject) => {
             const channel = new TestWebSocketChannel({ server, path: `${terminalsPath}/${terminalId}` });
@@ -109,12 +112,19 @@ describe('Task server / back-end', function (): void {
             channel.onMessage(msg => {
                 // check output of task on terminal is what we expect
                 const expected = `${isOSX ? 'tasking osx' : 'tasking'}... ${someString}`;
-                if (msg.toString().indexOf(expected) !== -1) {
+                // Instead of waiting for one message from the terminal, we wait for several ones as the very first message can be something unexpected.
+                // For instance: `nvm is not compatible with the \"PREFIX\" environment variable: currently set to \"/usr/local\"\r\n`
+                const currentMessage = msg.toString();
+                messages.unshift(currentMessage);
+                if (currentMessage.indexOf(expected) !== -1) {
                     resolve();
-                } else {
-                    reject(new Error(`expected sub-string not found in terminal output. Expected: "${expected}" vs Actual: "${msg.toString()}"`));
+                    channel.close();
+                    return;
                 }
-                channel.close();
+                if (messages.length >= messagesToWaitFor) {
+                    reject(new Error(`expected sub-string not found in terminal output. Expected: "${expected}" vs Actual messages: ${JSON.stringify(messages)}`));
+                    channel.close();
+                }
             });
         });
     });

--- a/packages/task/test-resources/task-long-running-osx
+++ b/packages/task/test-resources/task-long-running-osx
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 for i in {1..300}
 do

--- a/packages/task/test-resources/task-osx
+++ b/packages/task/test-resources/task-osx
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 for i in $@
 do

